src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPool.cs

```diff
@@ -0,0 +1,272 @@
+        {
+            if (ThreadPool.UseWindowsThreadPool)
+            {
+                WindowsThreadPool.GetAvailableThreads(out workerThreads, out completionPortThreads);
+            }
+            else
+            {
+                PortableThreadPool.ThreadPoolInstance.GetAvailableThreads(out workerThreads, out completionPortThreads);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static void NotifyWorkItemProgress()
+        {
+            if (ThreadPool.UseWindowsThreadPool)
+            {
+                WindowsThreadPool.NotifyWorkItemProgress();
+            }
+            else
+            {
+                PortableThreadPool.ThreadPoolInstance.NotifyWorkItemProgress();
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static bool NotifyWorkItemComplete(object threadLocalCompletionCountObject, int currentTimeMs) =>
+            ThreadPool.UseWindowsThreadPool ?
+            WindowsThreadPool.NotifyWorkItemComplete(threadLocalCompletionCountObject, currentTimeMs) :
+            PortableThreadPool.ThreadPoolInstance.NotifyWorkItemComplete(threadLocalCompletionCountObject, currentTimeMs);
+
+        internal static bool NotifyThreadBlocked() =>
+            ThreadPool.UseWindowsThreadPool ?
+            WindowsThreadPool.NotifyThreadBlocked() :
+            PortableThreadPool.ThreadPoolInstance.NotifyThreadBlocked();
+
+        internal static void NotifyThreadUnblocked()
+        {
+            if (ThreadPool.UseWindowsThreadPool)
+            {
+                WindowsThreadPool.NotifyThreadUnblocked();
+            }
+            else
+            {
+                PortableThreadPool.ThreadPoolInstance.NotifyThreadUnblocked();
+            }
+        }
+
+        /// <summary>
+        /// This method is called to request a new thread pool worker to handle pending work.
+        /// </summary>
+        internal static unsafe void RequestWorkerThread()
+        {
+            if (ThreadPool.UseWindowsThreadPool)
+            {
+                WindowsThreadPool.RequestWorkerThread();
+            }
+            else
+            {
+                PortableThreadPool.ThreadPoolInstance.RequestWorker();
+            }
+        }
+
+        internal static void ReportThreadStatus(bool isWorking)
+        {
+            Debug.Assert(!ThreadPool.UseWindowsThreadPool);
+            ReportThreadStatusCore(isWorking);
+        }
+
+        /// <summary>
+        /// Gets the number of thread pool threads that currently exist.
+        /// </summary>
+        /// <remarks>
+        /// For a thread pool implementation that may have different types of threads, the count includes all types.
+        /// </remarks>
+        public static int ThreadCount
+        {
+            get
+            {
+                return PortableThreadPool.ThreadPoolInstance.ThreadCount;
```

Comment:
All these methods should have `if (ThreadPool.UseWindowsThreadPool) ... Windows ... else ... Portable;`